lib/x/errors/too_many_requests.rb

```diff
@@ -1,23 +1,28 @@
 require_relative "client_error"
+require_relative "../rate_limit"
 
 module X
   class TooManyRequests < ClientError
-    def limit
-      response["x-rate-limit-limit"].to_i
-    end
-
-    def remaining
-      response["x-rate-limit-remaining"].to_i
+    def rate_limits
+      @rate_limits ||= limit_names.to_h do |limit|
+        [limit, RateLimit.new(limit, response)]
+      end
```

Comment:
Thinking some more about this, I’m tempted to only return rate limits with zero remaining requests here. Including others may cause confusion about the cause of the error. In that case, it would be unusual for there to be more than one element returned.

This class should also define a singular `rate_limit` instance method, which returns the rate limit with the farthest reset time.

Making these changes would dramatically simplify the `reset_at` definition.

Reply:
I did changes.